transport/transport.go

```diff
@@ -506,3 +507,50 @@ func wait(ctx context.Context, closing <-chan struct{}, proceed <-chan int) (int
 		return i, nil
 	}
 }
+
+const (
+	spaceByte   = byte(int(' '))
+	tildaByte   = byte(int('~'))
+	percentByte = byte(int('%'))
+)
+
+// matching https://github.com/grpc/grpc-java/pull/1517/files
+func grpcMessageEncode(grpcMessage string) string {
```

Comment:
grpcMessage is there to match that this is meant to be specifically for the `grpc-message` field - calling this `percentEncode` could actually be more confusing as when I hear that, I think URL encoding etc...this is something super custom (which again I don't like, but that's what it is).